src/polyfem/utils/ExpressionValue.cpp

```diff
@@ -181,13 +227,44 @@ namespace polyfem
 			tfunc_coo_ = coo;
 		}
 
+		void ExpressionValue::set_t(const json &t)
+		{
+			if (t.is_array())
+			{
+				for (int i = 0; i < t.size(); ++i)
+				{
+					t_index_[std::round(t[i].get<double>() * 1000.) / 1000.] = i;
+				}
+
+				if (mat_.size() != t_index_.size() && mat_expr_.size() != t_index_.size())
+					logger().error("Specifying varying dirichlet over time, however 'time_reference' does not match dirichlet boundary conditions.");
+			}
+		}
+
 		double ExpressionValue::operator()(double x, double y, double z, double t, int index) const
 		{
 			assert(unit_type_set_);
 
 			double result;
 			if (expr_.empty())
 			{
+				if (t_index_.size() > 0)
+				{
+					t = std::round(t * 1000.) / 1000.;
```

Comment:
what is this?

Reply:
This is the per timestep specification for boundary conditions. I think we agreed that it is brittle but we will refactor soon.